cloudnetwork/v1/types.go

```diff
@@ -0,0 +1,75 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudPrivateIPConfig performs assignments of a private IP address to the
+// primary NIC associated with cloud VMs. This is done by specifying the IP and
+// Kubernetes node which the IP should be assigned to. This CRD is intended to
+// be used by the network plugin which manages the cluster network. The spec
+// side represents the desired state requested by the network plugin, and the
+// status side represents the current state that this CRD's controller has
+// executed. No users will have permission to modify it, and if a cluster-admin
+// decides to edit it for some reason, their changes will be overwritten the
+// next time the network plugin reconciles the object. Note: the CR's name
+// must specify the requested private IP address (can be IPv4 or IPv6).
+// The IP format will be validated in an admission controller.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=cloudprivateipconfig,scope=Cluster
+type CloudPrivateIPConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
remove proto tags.  We cannot use them for CRs

Reply:
As I mentioned: https://github.com/openshift/api/pull/859#discussion_r607708231 I think this might confuse anyone doing `make generate-with-container` once this PR merges, since it will cause a bunch of diffs for this API.

Reply:
hrm.  weird....  I a problem for another day I suppose.